### PR TITLE
M2-4463: M2-4464: TransferOwnership and ShareApplet settings tests

### PR DESCRIPTION
**Pull request by wildkiomi**

### CODE REVIEW

File: src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+      expect(screen.getByTestId(`${dataTestid}-form`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-applet-name`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-keywords`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-agreement`)).toBeVisible();
+
+      fireEvent.click(screen.getByText('Share'));
+
+      await waitFor(() => {
+        expect(screen.getByTestId(`${dataTestid}-popup`)).toBeVisible();
```
Comment (alex-i-e) [inline src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx]: no need `await waitFor(...) wrapper` , use `await screen.find*` instead

File: src/shared/features/AppletSettings/TransferOwnershipSetting/TransferOwnershipSetting.test.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      const dataTestid = 'applet-settings-transfer-ownership';
+      renderWithProviders(<TransferOwnershipSetting />, { preloadedState, route, routePath });
+
+      expect(screen.getByTestId(`${dataTestid}-form`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-confirm`)).toBeVisible();
+
+      fireEvent.change(screen.getByLabelText(/Email/i), { target: { value: mockedEmail } });
+      fireEvent.click(screen.getByTestId(`${dataTestid}-confirm`));
+
+      await waitFor(() => expect(screen.getByTestId(`${dataTestid}-success-popup`)).toBeVisible());
```
Comment (alex-i-e) [inline src/shared/features/AppletSettings/TransferOwnershipSetting/TransferOwnershipSetting.test.tsx]: no need `await waitFor(...) wrapper` , use `await screen.find*` instead

File: src/shared/features/AppletSettings/TransferOwnershipSetting/TransferOwnershipSetting.test.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      ${`/builder/${mockedAppletId}/settings/${SettingParam.TransferOwnership}`}   | ${page.builderAppletSettingsItem} | ${'for builder'}
+    `('$description', async ({ route, routePath }) => {
+      mockAxios.post.mockResolvedValue(null);
+      const dataTestid = 'applet-settings-transfer-ownership';
+      renderWithProviders(<TransferOwnershipSetting />, { preloadedState, route, routePath });
+
+      expect(screen.getByTestId(`${dataTestid}-form`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-confirm`)).toBeVisible();
+
+      fireEvent.change(screen.getByLabelText(/Email/i), { target: { value: mockedEmail } });
```
Comment (alex-i-e) [inline src/shared/features/AppletSettings/TransferOwnershipSetting/TransferOwnershipSetting.test.tsx]: please use `userEvent` library for it, it's more closer to user interaction

File: src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      });
+    });
```
Comment (IhorNesterenko99) [inline src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx:115]: could you pls add also tests for the content of Success Modal, for click into Copy Applet Link on the Success modal, and for close Success modal?
image

File: src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      ${`/builder/${mockedAppletId}/settings/${SettingParam.ShareApplet}`}   | ${page.builderAppletSettingsItem} | ${'for builder'}
+    `('$description', async ({ route, routePath }) => {
+      const dataTestid = 'applet-settings-share-to-library';
+      renderWithProviders(<ShareAppletSetting />, { route, routePath, preloadedState });
+
+      expect(screen.getByTestId(`${dataTestid}-form`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-applet-name`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-keywords`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-agreement`)).toBeVisible();
+
```
Comment (IhorNesterenko99) [inline src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx:81]: Could you pls add check if checkbox was not checked, the Share button is disabled?

image

And also check if Applet name is already taken in the Library, there is an error:
image

File: src/shared/features/AppletSettings/TransferOwnershipSetting/TransferOwnershipSetting.test.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      ${`/dashboard/${mockedAppletId}/settings/${SettingParam.TransferOwnership}`} | ${page.appletSettingsItem}        | ${'for dashboard'}
+      ${`/builder/${mockedAppletId}/settings/${SettingParam.TransferOwnership}`}   | ${page.builderAppletSettingsItem} | ${'for builder'}
+    `('$description', async ({ route, routePath }) => {
+      mockAxios.post.mockResolvedValue(null);
+      const dataTestid = 'applet-settings-transfer-ownership';
+      renderWithProviders(<TransferOwnershipSetting />, { preloadedState, route, routePath });
+
+      expect(screen.getByTestId(`${dataTestid}-form`)).toBeVisible();
+      expect(screen.getByTestId(`${dataTestid}-confirm`)).toBeVisible();
+
```
Comment (IhorNesterenko99) [inline src/shared/features/AppletSettings/TransferOwnershipSetting/TransferOwnershipSetting.test.tsx:50]: could you pls add case if email is invalid?

image

Comment (nmialik) [reply src/shared/features/AppletSettings/TransferOwnershipSetting/TransferOwnershipSetting.test.tsx:50]: this case was covered by tests in another component

File: src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx (hunk 5 above)
Comment (nmialik) [reply src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx:81]: this case was covered by tests in another component

File: src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx (hunk 4 above)
Comment (nmialik) [reply src/shared/features/AppletSettings/ShareAppletSetting/ShareAppletSetting.test.tsx:115]: this case was covered by tests in another component

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**